tests/data_engine/test_querying.py

```diff
@@ -27,6 +28,42 @@ def test_simple_filter(ds):
     assert q.to_dict() == expected
 
 
+def test_versioning(ds):
+    add_int_fields(ds, "x")
+    add_int_fields(ds, "y")
+    add_int_fields(ds, "z")
+
+    ds2 = ((ds[ds[Field("x", as_of_time=123.99)] > 1]) &
+           (ds[ds[Field("x", as_of_time=345)] > 2]) |
+           (ds[ds[Field("y", as_of_time=789)] > 3])).\
+        select(Field("y", as_of_time=123), Field("x", as_of_time=456, alias="y_t1"), Field("z", as_of_time=dateutil.parser.parse("Wed 22 Nov")))
```

Comment:
This will need to be changed every year.
Hardcode one specific value with a boring `datetime.datetime()` constructor.
You can even do a [datetime.fromtimestamp](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromtimestamp) and reuse the same value as the query in the result (set the tz to `timezone.utc` too so there's no localtime issues)

Reply:
addes year